java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java

```diff
@@ -28,29 +45,51 @@ public String version() {
 
   @Override
   public void start(Map<String, String> props) {
-    log.info("Task initialising");
-    printAsciiHeader(jarManifest, "/azure-eventhubs-ascii.txt");
+    Optional<SourceTaskContext> contextOptional = ofNullable(context);
+    Map<String, String> configProps = null;
+    if (contextOptional.isPresent()) {
+      configProps = context.configs();
+      ofNullable(context.offsetStorageReader())
+          .ifPresent(TopicPartitionOffsetProvider::initialize);
+    }
 
-    Map<String, String> configProps;
-    boolean contextConfigEmpty = ofNullable(context).flatMap(
-        context -> ofNullable(context.configs()))
-      .flatMap(configs -> Optional.of(configs.isEmpty())).orElse(false);
-    if (contextConfigEmpty) {
+    if (configProps == null || configProps.isEmpty()) {
       configProps = props;
-    } else {
-      assert context != null;
-      configProps = context.configs();
     }
+
+    eventHubsKafkaConsumerController = new EventHubsKafkaConsumerController(
+        new AzureEventHubsConfig(configProps), new KafkaConsumerProvider(),
+        new ArrayBlockingQueue<>(BLOCKING_QUEUE_DEFAULT_CAPACITY));
+    String topic = configProps.get(AzureEventHubsConfigConstants.EVENTHUB_NAME);
+    initialize(eventHubsKafkaConsumerController, topic);
+  }
+
+  /**
+   * Initializes the Task. This method shouldn't be called if start() was already called with
+   * {@link EventHubsKafkaConsumerController} instance.
+   *
+   * @param eventHubsKafkaConsumerController {@link EventHubsKafkaConsumerController} for this task
+   * @param topic topic to subscribe to
+   */
+  public void initialize(EventHubsKafkaConsumerController eventHubsKafkaConsumerController,
+      String topic) {
+    this.eventHubsKafkaConsumerController = eventHubsKafkaConsumerController;
+    log.info("{} initialising.", getClass().getSimpleName());
+    printAsciiHeader(jarManifest, "/azure-eventhubs-ascii.txt");
```

Comment:
I am not sure if the ascii should be present in the task. 100 tasks, then there will be 100 ascii. Maybe we keep it just in the connector.

@davidsloan ?

Reply:
I moved it to Connector for the time being, I can move it back if we feel like it.

Reply:
I moved it to Connector for the time being, I can move it back if we feel like it.